### PR TITLE
Show stderr in the debugger

### LINKED ISSUE #86: Alfy eats errors from Run Script action
Alfred's workflow debugger only shows output from ```stderr```. These lines redirect all output from ```stderr``` to ```alfy.error()```, which then uses ```console.log()``` to print it to ```stdout```:

```
loudRejection(alfy.error);
process.on('uncaughtException', alfy.error);
hookStd.stderr(alfy.error);
```
This means that any errors that occur inside a Run Script action are never displayed in the workflow debugger.

### PR DESCRIPTION
**Pull request by LitoMore**
Resolve #86

---

<details>
<summary>
<b>IssueHunt Summary</b>
</summary>

### Referenced issues

This pull request has been submitted to:
- [#86: Alfy eats errors from Run Script action](https://issuehunt.io/repos/62888800/issues/86)
---

IssueHunt has been backed by the following sponsors. [Become a sponsor](https://issuehunt.io/membership/members)
</details>

### CODE REVIEW

File: test/test.js [hunk 1]
```diff
@@ -11,17 +10,6 @@ test('default', t => {
 	t.is(typeof alfyInstance.icon.error, 'string');
 });
 
-test.serial('.error()', async t => {
```
Comment (sindresorhus) [inline test/test.js]: Why are you removing the test? We still want to test `.error()`.

**Comment by sindresorhus** [comment]
Bump

**Comment by LitoMore** [comment]
Oh sorry, I missed this notification. I forgot why I removed this test. Let me check it.

**Comment by LitoMore** [comment]
Now I'm facing another bug. It seems Script Filter has the length limitation.

For example:

```javascript
alfy.output([{
	title: 'e'.repeat(1000),
	subtitle: 'Press ⌘L to see the full error and ⌘C to copy it.',
	valid: false,
	text: {
		copy: 'copy',
		largetype: 'stack'
	},
	icon: {
		path: exports.icon.error
	}
}]);
```

It will throw a JSON error like this:

```
[14:23:21.845] ERROR: alfred[Script Filter] JSON error: Unterminated string around character 30. in JSON:
{
	"items": [
		{
			"title": "eeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeee
```

We put too much letters to `title`, `copy` and `largetype`, they need to be simplified.

Should I do the simplify work in this pull request?

**Comment by sindresorhus** [comment]
Sure, what's the max length?

**Comment by LitoMore** [comment]
@sindresorhus The max length is 65532 (Alfred 4.0.6 [1124]).

**Comment by sindresorhus** [comment]
Truncate the `title` at 50000 then. Nobody will realistically have such long errors anyway.

**Comment by LitoMore** [comment]
Not only the title, `copy` and `largetype` also need to be truncated. 65532 is the max length of the whole JSON string.

**Comment by LitoMore** [comment]
Alfred will simplify the title like this below:

image

I think we could truncate the title at 100, truncate the `copy` and `largetype` at 20000 or smaller.

**Comment by sindresorhus** [comment]
`65532 - 50000 = 15532`, which is more than enough for the rest of the JSON, which is static.

**Comment by sindresorhus** [comment]
Ok, I now realize we're talking about different things. I was talking about the error specifically. You're talking about it generally.

`title` at 100 makes sense. `largetype` doesn't need that much. Let's cap it at 1000. The rest can be for the `copy`, which is the only thing that really needs to be long. We also need to documents these limits and why they are in place.

**Comment by sindresorhus** [comment]
Bump

**Comment by sindresorhus** [comment]
@LitoMore Bump

**Comment by LitoMore** [comment]
Roger. I will follow up on the weekend.

**Comment by sindresorhus** [comment]
Bump

**Pull request closed without merging**